### PR TITLE
Remove useless promise and add some error handling

### PR DESCRIPTION
**Pull request by cjh980402**
Optimze some code and add error http error handling of `downloadProgressive`.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -166,7 +166,7 @@ export class Client {
 export class Downloader {
 	constructor();
 
-	downloadHLS(url: string, options?: m3u8Options): Promise<m3u8Stream>;
+	downloadHLS(url: string, options?: m3u8Options): m3u8Stream;
```
Comment (twlite) [inline index.d.ts:169]: This was `Promise<m3u8Stream>` to align with `progressive` stream which is `Promise<IncomingMessage>`

Comment (cjh980402) [reply index.d.ts:169]: But it doesn't actually need a promise in Downloader.
However `Song.downloadHLS` is still return Promise<m3u8Stream>.

I am sorry to late review. I wrote it yesterday but it was pending state. :(

Comment (twlite) [reply index.d.ts:169]: we could also update `downloadProgressive` to not return a promise by doing something like:

```js
const outStream = new PassThrough();

setImmediate(() => {
    req.get(url, res => {
        res.pipe(outStream);
    });
});

return outStream;
```

Comment (cjh980402) [reply index.d.ts:169]: I see.
How about other code in my PR? Is it OK?

Comment (cjh980402) [reply index.d.ts:169]: Also we should handle 'error' event of req.get's return value.

Comment (twlite) [reply index.d.ts:169]: > Also we should handle 'error' event of req.get's return value.

Instead, we should forward `error` to `outStream` and let users handle it

Comment (cjh980402) [reply index.d.ts:169]: That's quite good way. Then, Will you change it from this PR to that way?

**Pull request closed without merging**